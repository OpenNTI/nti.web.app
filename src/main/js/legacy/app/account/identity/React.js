const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
