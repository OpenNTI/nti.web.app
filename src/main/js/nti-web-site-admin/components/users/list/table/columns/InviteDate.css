.cell {
	composes: veridically-center from './column.css';
	font-size: 14px;
	color: var(--secondary-grey);
}
