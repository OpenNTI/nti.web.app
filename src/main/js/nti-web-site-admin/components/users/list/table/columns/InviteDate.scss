@import '~@nti/style-common';

.users-table-container {
	.invitedate-col {
		.cell {
			font-size: 14px;
			color: var(--secondary-grey);
			line-height: 60px;
		}
	}
}
