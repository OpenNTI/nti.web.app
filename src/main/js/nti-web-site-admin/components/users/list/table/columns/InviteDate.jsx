import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';
import {DateTime} from '@nti/web-commons';
import cx from 'classnames';

import styles from './InviteDate.css';

const t = scoped('nti-web-site-admin.components.users.list.table.columns.InviteDate', {
	headerTitle: 'Invite Date',
	title: 'Invite Date',
	today: 'Today',
	yesterday: 'Yesterday'
});

export default class InviteDate extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired
	}

	static cssClassName = 'invite-date-column';

	static Name = () => t('title')

	static SortKey = 'created_time';

	render () {
		const {item} = this.props;

		const isToday = DateTime.isToday(item.getCreatedTime());

		let newDate = new Date(item.getCreatedTime());
		newDate.setDate(newDate.getDate() + 1);
		const isYesterday = DateTime.isToday(newDate);

		return (
			<div className={cx('cell', styles.cell)}>
				{!isToday && !isYesterday && <DateTime className={styles.value} date={item.getCreatedTime()} format="LL" />}
				{isToday && <span>{t('today')}</span>}
				{isYesterday && <span>{t('yesterday')}</span>}
			</div>
		);
	}
}
