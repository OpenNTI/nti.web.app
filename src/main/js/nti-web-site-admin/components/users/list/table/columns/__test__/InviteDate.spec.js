/* eslint-env jest */
import React from 'react';
import { create, act } from 'react-test-renderer';

import InviteDate from '../InviteDate';

describe('Site admin user table invite date column test', () => {
	const verifyColumn = date => {
		let cmp;
		act(() => {
			cmp = create(
				<InviteDate
					item={{
						getCreatedTime: () => {
							return date;
						},
					}}
				/>
			);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	};

	test('Test date is now', () => {
		// label should say "Today"
		verifyColumn(Date.now());
	});

	test('Test date is yesterday', () => {
		// label should say "Yesterday"
		verifyColumn(new Date(Date.now() - 1000 * 60 * 60 * 24));
	});

	test('Test date is many days ago', () => {
		// label should show the formatted date as "Oct 31, 2017"
		verifyColumn(new Date('10/31/2017'));
	});
});
