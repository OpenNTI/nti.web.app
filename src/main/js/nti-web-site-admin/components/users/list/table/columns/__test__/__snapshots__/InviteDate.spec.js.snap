// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user table invite date column test Test date is many days ago 1`] = `
<div
  className="cell"
>
  <time
    dateTime="2017-10-31T00:00:00+00:00"
  >
    October 31, 2017
  </time>
</div>
`;

exports[`Site admin user table invite date column test Test date is now 1`] = `
<div
  className="cell"
>
  <span>
    Today
  </span>
</div>
`;

exports[`Site admin user table invite date column test Test date is yesterday 1`] = `
<div
  className="cell"
>
  <span>
    Yesterday
  </span>
</div>
`;
